### PR TITLE
Object 3d refactoring.

### PR DESCRIPTION
**Pull request by maxcapodi78**
made hidden some private methods (like m_Editor)
added property objects_by_names since most of users knows the object name and not the id

### CODE REVIEW

File: pyaedt/modeler/cad/Primitives.py [hunk 1]
```diff
@@ -1372,12 +1370,25 @@ def refresh(self):
...
+    @property
+    def objects_by_name(self):
+        """Object dictionary organized by name.
+
+        Returns
+        -------
+        dict
+        """
+        obj_dict = {}
+        for _, v in self.objects.items():
```
Comment (Alberto-DM) [inline pyaedt/modeler/cad/Primitives.py:1388]: why not like this?

```suggestion
        for v in self.objects.values():
```

File: pyaedt/modeler/cad/Modeler.py [hunk 2]
```diff
@@ -2526,15 +2495,11 @@ def convert_to_selections(self, object_id, return_list=False):
...
-        elif not isinstance(object_id, list):
+        if not isinstance(object_id, list):
             object_id = [object_id]
         objnames = []
         for el in object_id:
-            if isinstance(el, int) and el in list(self.objects.keys()):
+            if isinstance(el, int) and el in self.objects:
                 objnames.append(self.objects[el].name)
             elif isinstance(el, int):
                 objnames.append(el)
```
Comment (Alberto-DM) [inline pyaedt/modeler/cad/Modeler.py:2505]: why there is this case? The returned list should contain the object names, here it is putting in an id.

File: pyaedt/modeler/cad/Primitives.py [hunk 3]
```diff
@@ -1310,8 +1309,8 @@ def get_obj_id(self, objname):
             Object ID.
 
         """
-        if objname in self.object_id_dict:
-            return self.object_id_dict[objname]
+        if objname in self._object_id_to_names:
+            return self._object_id_to_names[objname]
```
Comment (Alberto-DM) [inline pyaedt/modeler/cad/Primitives.py:1313]: Here we pass the name as argument and we get back the id. So, should the name be 'object_name_to_id'?
I mean, the dictionary goes from name to id. Am I wrong?

**Review by Alberto-DM — APPROVED** [review]
Just a couple of comments. Otherwise LGTM.

File: pyaedt/modeler/cad/Primitives.py [hunk 4]
```diff
@@ -592,11 +592,10 @@ def does_object_exists(self, object):
             else:
                 return False
```
Comment (MaxJPRey) [inline pyaedt/modeler/cad/Primitives.py:593]:
```suggestion
```

File: pyaedt/modeler/cad/Primitives.py [hunk 5]
```diff
@@ -592,11 +592,10 @@ def does_object_exists(self, object):
...
         else:
-            for el in self.objects:
-                if self.objects[el].name == object:
-                    return True
-
-        return False
+            if object in self.objects_by_name:
+                return True
+            else:
+                return False
```
Comment (MaxJPRey) [inline pyaedt/modeler/cad/Primitives.py:598]:
```suggestion
        return False
```

File: pyaedt/modeler/cad/Primitives.py [hunk 6]
```diff
@@ -592,11 +592,10 @@ def does_object_exists(self, object):
             else:
                 return False
         else:
```
Comment (MaxJPRey) [inline pyaedt/modeler/cad/Primitives.py:594]: Here the entire code:
```
        if isinstance(object, int):
            if object in self.objects:
                return True
            else:
                return False
        else:
            if object in self.objects_by_name:
                return True
            else:
                return False
```

Could become:
  ```
      if isinstance(object, int):
            if object in self.objects:
                return True
            elif object in self.objects_by_name:
                return True
            return False
```

Comment (maxcapodi78) [reply pyaedt/modeler/cad/Primitives.py:594]: thanks. Improved

File: pyaedt/modeler/cad/Primitives.py (hunk 1 above)
Comment (maxcapodi78) [reply pyaedt/modeler/cad/Primitives.py:1384]: because for Ironpython I should use list(self.objects.values()). I prefer this approach

**Pull request merged**